pinot-broker/src/main/java/org/apache/pinot/broker/requesthandler/BaseBrokerRequestHandler.java

```diff
@@ -202,24 +208,21 @@ private BrokerResponseNative handleRequest(long requestId, String query,
       throws Exception {
     LOGGER.debug("SQL query for request {}: {}", requestId, query);
 
-    // Compile the request
-    long compilationStartTimeNs = System.nanoTime();
+    long compilationStartTimeNs;
     PinotQuery pinotQuery;
     try {
-      if (sqlNodeAndOptions != null) {
-        // Include parse time when the query is already parsed
-        compilationStartTimeNs -= sqlNodeAndOptions.getParseTimeNs();
-      } else {
-        sqlNodeAndOptions = CalciteSqlParser.compileToSqlNodeAndOptions(query);
-      }
+      // Parse the request
+      sqlNodeAndOptions = sqlNodeAndOptions != null ? sqlNodeAndOptions
+          : RequestUtils.parseQuery(requestId, query, request);
+      // Compile the request into PinotQuery
+      compilationStartTimeNs = System.nanoTime();
```

Comment:
I feel this will be incorrect accounting as it will miss out the time taken to build the sqlNodeAndOptions. Shouldn't we stick to what the current code is doing ?

```
if (sqlNodeAndOptions != null) {
        // Include parse time when the query is already parsed
        compilationStartTimeNs -= sqlNodeAndOptions.getParseTimeNs();
      }
```

Reply:
no. the reporting is now using compilationTime + parsingTime as an overall reporting